components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/responsetime_util.bal

```diff
@@ -149,29 +153,59 @@ public function generateRequestResponseExecutionDataEvent(http:Response response
     }
     //request method
-    requestResponseExecutionDTO.apiMethod = <string>context.attributes[API_METHOD_PROPERTY];
-    int initTime = <int>context.attributes[REQUEST_TIME];
-    int timeRequestOut = <int>invocationContext.attributes[TS_REQUEST_OUT];
-    int timeResponseIn = <int>invocationContext.attributes[TS_RESPONSE_IN];
-    requestResponseExecutionDTO.serviceTime = timeRequestOut - initTime;
-    requestResponseExecutionDTO.backendTime = timeResponseIn - timeRequestOut;
-    requestResponseExecutionDTO.responseTime = timeResponseIn - initTime;
+    if (context.attributes[API_METHOD_PROPERTY] is string) {
+        requestResponseExecutionDTO.apiMethod = <string>context.attributes[API_METHOD_PROPERTY];
+    }
+    
+    if (context.attributes[REQUEST_TIME] is int && invocationContext.attributes[TS_RESPONSE_IN] is int && invocationContext.attributes[TS_REQUEST_OUT] is int) {
+        int initTime = <int>context.attributes[REQUEST_TIME];
+        int timeRequestOut = <int>invocationContext.attributes[TS_REQUEST_OUT];
+        int timeResponseIn = <int>invocationContext.attributes[TS_RESPONSE_IN];
+        requestResponseExecutionDTO.serviceTime = timeRequestOut - initTime;
+        requestResponseExecutionDTO.backendTime = timeResponseIn - timeRequestOut;
+        requestResponseExecutionDTO.responseTime = timeResponseIn - initTime;
+    }
+    
     //dummy values for protocol and destination for now
-    requestResponseExecutionDTO.protocol = <string>context.attributes[PROTOCOL_PROPERTY];
-    requestResponseExecutionDTO.destination = <string>invocationContext.attributes[DESTINATION];
+    if (context.attributes[PROTOCOL_PROPERTY] is string) {
+        requestResponseExecutionDTO.protocol = <string>context.attributes[PROTOCOL_PROPERTY];
+    }
+    if (invocationContext.attributes[DESTINATION] is string) {
+        requestResponseExecutionDTO.destination = <string>invocationContext.attributes[DESTINATION];
+    }
 
     //Set data which were set to context in the Request path
-    requestResponseExecutionDTO.applicationOwner = <string>context.attributes[APPLICATION_OWNER_PROPERTY];
-    requestResponseExecutionDTO.apiCreatorTenantDomain = <string>context.attributes[API_CREATOR_TENANT_DOMAIN_PROPERTY];
-    requestResponseExecutionDTO.apiTier = <string>context.attributes[API_TIER_PROPERTY];
-    requestResponseExecutionDTO.throttledOut = <boolean>context.attributes[CONTINUE_ON_TROTTLE_PROPERTY];
-    requestResponseExecutionDTO.userAgent = <string>context.attributes[USER_AGENT_PROPERTY];
-    requestResponseExecutionDTO.userIp = <string>context.attributes[USER_IP_PROPERTY];
-    requestResponseExecutionDTO.requestTimestamp = <int>context.attributes[REQUEST_TIME_PROPERTY];
+    if (context.attributes[APPLICATION_OWNER_PROPERTY] is string) {
+        requestResponseExecutionDTO.applicationOwner = <string>context.attributes[APPLICATION_OWNER_PROPERTY];
+    }
+    
```

Comment:
Lets remove empty line

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/1319/commits/b694c6960318d34763d343da195320faf623a55f